tests/test_site_apportionment/test_site_apportionment.py

```diff
@@ -634,3 +636,127 @@ def test_sort_rows_missing_cols(self, caplog):
 
         with pytest.raises(KeyError):
             sort_rows_order_cols(input_df, cols_in_order)
+
+
+class TestWeightValues(object):
+    """Tests for weight_values."""
+    
+    @pytest.fixture(scope="function")
+    def weight_values_test_df(self):
+        """Test data for weight_values."""
+        frame = {
+            "reference": [0, 1, 2, 3, 4],
+            "val_col_1": [10, 14, 16, 18, 6],# test using ints rather than float
+            "val_col_2": [3.5, 4.5, 10, np.nan, 0],# test np.nan+0+floats
+            "weight_col": [1.5, 1.5, 1.0, 2.0, 3.0]
+                }
+        return pandasDF(frame)
+    
+    @pytest.fixture(autouse=True)
+    def set_attrs(self):
+        """Set class attributes to pass to function."""
+        self.value_cols = ["val_col_1", "val_col_2"]
+        self.weight_col = "weight_col"
+
+    def test_weight_values_raises(self, weight_values_test_df):
+        """Tests for raises in weight_values."""
+        with pytest.raises(KeyError, match=".*not_weight_col.*"):
+            weight_values(weight_values_test_df, 
+                          self.value_cols, 
+                          "not_weight_col")
+
+    def test_weight_values_on_pass(self, weight_values_test_df):
+        """General tests for weight_values."""
+        output = weight_values(weight_values_test_df,
+                                self.value_cols, 
+                                self.weight_col)
+        assert np.array_equal(output["val_col_1"],
+                       [15.0, 21.0, 16.0, 36.0, 18.0])
+        assert np.array_equal(output["val_col_2"],
+                       [5.25, 6.75, 10.0, np.nan, 0.0], equal_nan=True)
+        
+
+class TestCreateCategoryDf(object):
+    """Tests for create_category_df."""
+
+    @pytest.fixture(scope="function")
+    def category_df_input(self):
+        """Test data for create_category_df."""
+        data = [[49900000404, 0.0, 202212, 'R', 'AA', np.nan, np.nan, np.nan, 100.0, np.nan],
+                [49900000404, 1.0, 202212, 'R', 'AA', 'C', 29.0, np.nan, 100.0, 243600.0379],
+                [49900000404, 2.0, 202212, 'R', 'AA', 'D', 29.0, np.nan, 100.0, 0.0],
+                [49900000408, 0.0, 202212, 'R', 'I', np.nan, 32.0, np.nan, 100.0, np.nan],
+                [49900000408, 1.0, 202212, 'R', 'AA', 'C', 33.0, np.nan, 100.0, 0.0],
+                [49900000407, 2.0, 202212, 'test', 'AA', 'D', np.nan, np.nan, 100.0, 0.0],
+                [49900000576, 1.0, 202212, 'TMI', 'AA', 'C', np.nan, np.nan, np.nan, 0.0],
+                [49900000960, 0.0, 202212, 'R', 'AA', np.nan, 1.0, np.nan, np.nan, np.nan],
+                [49900000960, 1.0, 202212, 'R', 'AA', 'C', 23.0, np.nan, np.nan, np.nan],
+                [49900001029, 1.0, 202212, 'TMI', 'I', 'C', 5.0, np.nan, np.nan, 833.3333]]
+        
+        columns = ['reference', 'instance', 'period', 'imp_marker', '201', 
+                   '200', 'pg_numeric', '601', '602', '210']
```

Comment:
please could you update to our style of having the columns before the data, though I don't mind if they're not all on different lines.